File: PHP/FindValidMatrixGivenRowAndColumnSumsTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit FindValidMatrixGivenRowAndColumnSumsTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("FindValidMatrixGivenRowAndColumnSums.php");

class FindValidMatrixGivenRowAndColumnSumsTest extends TestCase {

    public function testFindValidMatrixGivenRowAndColumnSums() {
        $obj = new FindValidMatrixGivenRowAndColumnSums();
        //Test 1
        $want = [[3,0],[1,7]];
        $got = $obj->restoreMatrix( [3,8], [4,7] );
        echo "\nTest 1: retornou " . var_export($got, true) . " == esperado: " . var_export($want, true);
        $this->assertEquals($want, $got);
        //Test 2
        $want = [[0,5,0],[6,1,0],[2,0,8]];
        $got = $obj->restoreMatrix( [5,7,10], [8,6,8] );
        echo "\nTest 2: retornou " . var_export($got, true) . " == esperado: " . var_export($want, true);
        $this->assertEquals($want, $got);
    }
}

File: GO/FindValidMatrixGivenRowAndColumnSums_test.go
/**
 * @version GO 1.22.3
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestFindValidMatrixGivenRowAndColumnSums
 */
package GO

import (
	"reflect"
	"testing"
)

func TestFindValidMatrixGivenRowAndColumnSums(t *testing.T) {
	tables := []struct {
		want   [][]int
		rowSum []int
		colSum []int
	}{
		{[][]int{{1, 2}, {0, 5}}, []int{3, 8}, []int{1, 7}},
		{[][]int{{5, 0, 0}, {3, 4, 0}, {0, 2, 8}}, []int{5, 7, 10}, []int{8, 6, 8}},
	}

	for _, table := range tables {
		got := restoreMatrix(table.rowSum, table.colSum)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
